### ISSUE TITLE
Is Llama 2 compatible for finetuning using lora?

### ISSUE THREAD

**Issue by palash04** [mosaicml/llm-foundry#482]
I am trying to finetune llama2 using lora.

Following are the changes I made in yaml
```
model:
  name: hf_causal_lm
  pretrained_model_name_or_path: meta-llama/Llama-2-7b-hf
  init_device: mixed
  pretrained: true
  tie_word_embeddings: true
  use_auth_token: true 

# LORA
lora:
  args:
    r: 16
    lora_alpha: 32
    lora_dropout: 0.05
    target_modules: ['Wqkv']

# Tokenizer
tokenizer:
  name: meta-llama/Llama-2-7b-hf
  kwargs:
    model_max_length: ${max_seq_len}
```

Getting following error:
You are using a model of type llama to instantiate a model of type mpt. This is not supported for all configurations of models and can yield errors.

<img width="707" alt="Screenshot 2023-07-22 at 6 35 26 PM" src="https://github.com/mosaicml/llm-foundry/assets/26361028/f031f1bb-716c-4e84-bf5e-d10f3041ca82">

**Comment by creatorrr** [comment]
Not working for me :(

**Comment by creatorrr** [comment]
@danbider @dakinggg any pointers on how I can take a crack at implementing this?

**Comment by palash04** [comment]
@creatorrr I think [435](https://github.com/mosaicml/llm-foundry/pull/435) here tries to solve this, but have no idea when it will be merged. Looks like it is failing some checks for python 3.8, and 3.9. Passed for 3.10

**Comment by palash04** [comment]
@creatorr, this is being fixed [here - 435](https://github.com/mosaicml/llm-foundry/pull/435)

**Comment by dakinggg** [comment]
Closed by #886, this should work now :)
